data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/IngressProducerReconcilableStore.java

```diff
@@ -78,10 +81,20 @@ public IngressProducerReconcilableStore(
     this.ingressInfos = new HashMap<>();
     this.producerReferences = new HashMap<>();
     this.pathMapper = new HashMap<>();
+    this.hostMapper = new HashMap<>();
   }
 
-  public IngressProducer resolve(String path) {
-    return pathMapper.get(path);
+  public IngressProducer resolve(String host, String path) {
+    // Ignore the host when there's a path given in the request.
+    // That means, we support these modes:
+    // - Request coming to "/path" --> path is used for matching
+    // - Request coming to "/" --> hostname is used for matching
+    if (pathMapper.get(path) != null) {
+      return pathMapper.get(path);
+    } else if (hostMapper.get(host) != null) {
+      return hostMapper.get(host);
+    }
+    return null;
```

Comment:
else after return, we can simplify since a map without a key returns `null` and be more consistent with the codebase with:

```suggestion
    if (pathMapper.get(path) != null) {
      return pathMapper.get(path);
    } 
    return hostMapper.get(host);
```